### PR TITLE
Hi-performance metadata API endpoint changes part 1

### LINKED ISSUE #9781: Old Postgres scheme is not removed after "sync db scheme"
Bug report:
I am using metabase.jar 0.32.4 with H2 internal database on Windows. My analytics data is located on a Postgres DB. I have some old schemes on Postgres which do not exist anymore, but metabase shows them since long times ago it has seen them and "Sync database schema now" has no effect on it.

### PR DESCRIPTION
**Pull request by camsaul**
*  `GET /api/database/:id/schemas` should work for saved questions virtual DB
*  `GET /api/database/:id/schema/:schema` should work for saved questions virtual DB
*  Fixes #9781

### CODE REVIEW

File: src/metabase/api/database.clj [hunk 1]
```diff
@@ -576,15 +584,25 @@
   "Returns a list of all the schemas found for the database `id`"
   [id]
   (api/read-check Database id)
-  (->> (db/select-field :schema Table :db_id id)
+  (->> (db/select-field :schema Table :db_id id {:order-by [[:%lower.schema :asc]]})
```
Comment (flamber) [inline src/metabase/api/database.clj]: Would adding `:active true` here also fix #9781

Comment (camsaul) [reply src/metabase/api/database.clj]: good catch @flamber, I'll make that change 👍

Comment (flamber) [reply src/metabase/api/database.clj]: All credit to @lindsay-stevens - I was just reading your code and made 1+1 :wink:

Comment (lindsay-stevens) [reply src/metabase/api/database.clj]: thank you both!

**Pull request merged**